code/modules/client/client procs.dm

```diff
@@ -109,135 +109,53 @@
 			if("shop")
 				if(href_list["KarmaBuy"])
 					var/karma=verify_karma()
-					switch(href_list["KarmaBuy"])
-						if("1")
-							if(karma <5)
-								to_chat(usr, "You do not have enough karma!")
+					if(!isnull(karma)) //Doesn't display anything if karma database is down.
```

Comment:
I think you could replace this with an early return. e.g.
`
if(isnull(karma))
 return
`